### ISSUE TITLE
SAM: better support for API-based lambdas

### ISSUE THREAD

**Issue by JadenSimon** [aws/aws-toolkit-vscode#2455]
Parent issue to track features or improvements regarding API-based lambdas.

#### Proposed new features:
* `interactive` mode 
  * #1840
  * Start a server without a payload
  * Toolkit attaches a debugger whenever SAM CLI executes a function
  * Persist the server until the user closes out the session 
* Support more options (port, logging, host, 'warm containers', etc.) 
  * #2446
  * Toolkit may need to explicitly handle these options to work correctly
* A 'workbench' UI for sending payloads to the local API (to pair with `interactive` mode)
  * Could either be a notebook or tacked onto the SAM config webview 
* More API Gateway resources in templates (e.g. V2)
  * #1489
  * AWS::Serverless::HttpApi

#### General improvements:
* Uniform timeouts
  * Toolkit currently polls for a set amount of time to wait for the server to start
* Better discoverability
  * It's not obvious how to configure things for anything more complex than a 'HelloWorld' project
* Better logging during executions (less noise)
* Fail faster
  * Improve the experience by finding common issues with configurations (wrong path, wrong method, etc.)

**Comment by justinmk3** [comment]
> Uniform timeouts

We could have very large timeouts if the Toolkit correctly handled VSCode debugger "Stop" button (assuming VSCode gives us a way to do this). Or as a workaround we could present a progress status w/ "Cancel" button.

**Comment by JadenSimon** [comment]
Yeah, many of the timeouts seem to be more like workarounds rather than features. We still want to know if a process is hanging, but I think it's best if we give the user the final choice in cancelling an operation. The automatic cancellation has caused a lot of issues.

**Comment by justinmk3** [comment]
@bryantqo I added some notes to #2535 to try to capture your idea.
